manifests/resource/map.pp

```diff
@@ -65,7 +65,10 @@
 #
 define nginx::resource::map (
   String[2] $string,
-  Variant[Array, Hash] $mappings,
+  Variant[
+    Array[Struct[{ 'key' => String, 'value' => Any }]],
+    Hash[String, Any]
+  ] $mappings = [],
```

Comment:
should we introduce a custom datatype for this ? my personal rule of thumb is: do I need newlines in a datatype? worth migrating it out of the class.

Reply:
I guess yes, it's good time to make another type for this

Reply:
The only concerning question is.. do we need to reuse the type in the EPP also (and move Hash -> Array conversion there maybe)

Reply:
Added Nginx::StringMappings and moved Hash -> Array conversion to the EPP